Sources/SafeDICore/Generators/DependencyTreeGenerator.swift

```diff
@@ -256,9 +263,17 @@ public final class DependencyTreeGenerator {
             receivableProperties: Set<Property>,
             instantiables: OrderedSet<Instantiable>
         ) {
+            let forwardedProperties = Set(
+                scope
+                    .instantiable
+                    .dependencies
+                    .filter(\.isForwarded)
+                    .map(\.property)
+            )
             for receivedProperty in scope.receivedProperties {
                 let parentContainsProperty = receivableProperties.contains(receivedProperty)
-                if !parentContainsProperty {
+                let propertyIsForwarded = forwardedProperties.contains(receivedProperty)
+                if !parentContainsProperty && !propertyIsForwarded {
```

Comment:
this is the crux of the change that makes the new tests pass. We were already inspecting properties that were forwarded by a higher-up scope (they're in `receivedProperties`), but we weren't inspecting the properties in _out_ scope.